src/main/java/org/openstreetmap/atlas/checks/commands/MapRouletteUploadCommand.java

```diff
@@ -0,0 +1,202 @@
+     * a particular project name.
+     */
+    private class TaskDeserializer implements JsonDeserializer<Task>
+    {
+        private static final String PROPERTIES = "properties";
+        private static final String GENERATOR = "generator";
+        private static final String INSTRUCTIONS = "instructions";
+        private static final String ID = "id";
+        private static final String FEATURES = "features";
+        private String projectName;
+
+        TaskDeserializer(final String projectName)
+        {
+            this.projectName = projectName;
+        }
+
+        @Override public Task deserialize(final JsonElement json, final Type typeOfT,
+                final JsonDeserializationContext context) throws JsonParseException
+        {
+            final JsonObject full = json.getAsJsonObject();
+            final JsonObject properties = full.get(PROPERTIES).getAsJsonObject();
+            final String challengeName = properties.get(GENERATOR).getAsString();
+            final String instruction = properties.get(INSTRUCTIONS).getAsString();
+            final String taskID = properties.get(ID).getAsString();
+            final JsonArray geojson = full.get(FEATURES).getAsJsonArray();
+            final Task result = new Task();
+            result.setChallengeName(challengeName);
+            // Note: since the points are contained inside the geojson feature, and we're
+            // re-serializing on our way to MapRoulette, we don't need to explicitly parse out the
+            // points from the geojson. If you're considering pulling this class out for more
+            // general use, points should be parsed as well.
+            result.setGeoJson(Optional.of(geojson));
+            result.setInstruction(instruction);
+            result.setProjectName(projectName);
+            result.setTaskIdentifier(taskID);
+            return result;
+        }
+    }
+
+    private static final Switch<File> INPUT_DIRECTORY = new Command.Switch<>("logfiles",
+            "Path to folder containing log files to upload to MapRoulette.", File::new, Optionality.REQUIRED);
+    private static final Switch<File> CONFIG_LOCATION = new Command.Switch<>("config",
+            "Path to a file containing MapRoulette challenge configuration.", File::new, Optionality.REQUIRED);
+    private static final String PARAMETER_CHALLENGE = "challenge";
+    private static final String LOG_EXTENSION = "log";
+    private static final Logger logger = LoggerFactory.getLogger(MapRouletteUploadCommand.class);
+    private final Map<String, Challenge> checkNameChallengeMap;
+
+    public MapRouletteUploadCommand()
+    {
+        super();
+        this.checkNameChallengeMap = new HashMap<>();
+    }
+
+    @Override protected void execute(final CommandMap commandMap, final MapRouletteConfiguration configuration)
+    {
+        final File[] files = ((File) commandMap.get(INPUT_DIRECTORY)).listFiles();
```

Comment:
You don't need to explicitly check for input files. You can let the Switch class handle that

Reply:
I've changed this up to be a little cleaner, but I'm not sure it's the way you wanted. Let me know if you'd like it to be different